client/my-sites/plans-features-main/products-header.jsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ExternalLinkWithTracking from 'components/external-link/with-tracking';
+import FormattedHeader from 'components/formatted-header';
+import { getSiteSlug } from 'state/sites/selectors';
+import { getSitePurchases } from 'state/purchases/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isJetpackBackup } from 'lib/products-values';
+import { JETPACK_BACKUP_PRODUCT_LANDING_PAGE_URL } from 'lib/products-values/constants';
+import { addQueryArgs } from 'lib/route';
+import QueryPlans from 'components/data/query-plans';
+import QuerySitePlans from 'components/data/query-site-plans';
+import QuerySitePurchases from 'components/data/query-site-purchases';
+
+class PlansFeaturesMainProductsHeader extends Component {
+	siteHasJetpackBackup() {
+		const { purchases } = this.props;
+
+		if ( isEmpty( purchases ) ) {
+			return false;
+		}
+
+		// Search through purchased products for Jetpack Backup.
+		return ! isEmpty(
+			purchases.find( purchase => purchase.active && isJetpackBackup( purchase ) )
+		);
+	}
+
+	getSubHeader() {
+		const { siteSlug, translate } = this.props;
+		const baseCopy = translate( "Just looking for a backups? We've got you covered." );
+
+		// Don't render a link if a user already has a Jetpack Backup product or a plan with a backup feature.
+		if ( this.siteHasJetpackBackup() ) {
+			return baseCopy;
+		}
+
+		// If we have a site in this context, add it to the landing page URL.
+		const linkUrl = siteSlug
+			? addQueryArgs( { site: siteSlug }, JETPACK_BACKUP_PRODUCT_LANDING_PAGE_URL )
+			: JETPACK_BACKUP_PRODUCT_LANDING_PAGE_URL;
+
+		return (
+			<Fragment>
+				{ baseCopy }
+				<br />
+				<ExternalLinkWithTracking
+					href={ linkUrl }
+					tracksEventName="calypso_plan_link_click"
+					tracksEventProps={ {
+						link_location: 'product_jetpack_backup_description',
+						link_slug: 'which-one-do-i-need',
+					} }
+					icon
+				>
+					{ translate( 'Which backup option is best for me?' ) }
+				</ExternalLinkWithTracking>
+			</Fragment>
+		);
+	}
+
+	render() {
+		const { selectedSiteId, translate } = this.props;
+
+		return (
+			<Fragment>
+				<QueryPlans />
+				<QuerySitePlans siteId={ selectedSiteId } />
+				<QuerySitePurchases siteId={ selectedSiteId } />
+				<FormattedHeader
+					headerText={ translate( 'Solutions' ) }
+					subHeaderText={ this.getSubHeader() }
+					compactOnMobile
+					isSecondary
+				/>
+			</Fragment>
+		);
+	}
+}
+
+PlansFeaturesMainProductsHeader.propTypes = {
+	purchases: PropTypes.array,
+	selectedSiteId: PropTypes.number,
+	siteSlug: PropTypes.string,
+};
+
+export default connect( ( state, { siteId } ) => {
```

Comment:
Seems like in this component we never use the `siteId` prop. Should we completely remove it and rely on the `selectedSiteId` one instead?

Reply:
I thought about it but in the end I took [the same approach as in the `ProductsSelector`](https://github.com/Automattic/wp-calypso/blob/7b1074f1119fdff6ce182df8f3349c2d1da49f27/client/blocks/product-selector/index.jsx#L583-L584).

I guess it is a bit more flexible, though we don't really use it at this point.

Reply:
There's the difference though. For `ProductsSelector` we use that ID (even if it's only for devdocs). And for this component we don't. So I'd suggest going YAGNI and cleaning it up.